### PR TITLE
fix: New long tail assets configuration

### PR DESCRIPTION
**Pull request by akhercha**
Resolves: NA

## Updated
- Configuration for long tail assets

## TODO
- [ ] Medium alerts for individual source deviation

### CODE REVIEW

File: src/processing/spot.rs [hunk 1]
```diff
@@ -217,53 +218,83 @@ pub async fn process_long_tail_asset(
...
-    for (i, (src1, price1)) in prices.iter().enumerate() {
-        for (src2, price2) in prices.iter().skip(i + 1) {
-            let deviation = (price1 - price2).abs() / price1;
-            LONG_TAIL_ASSET_DEVIATION
-                .with_label_values(&[network_env, &pair, data_type, src1, src2])
-                .set(deviation);
+            let (source_deviation, _) = source_deviation(&data, normalized_price).await?;
+            Ok((time, source_deviation))
         }
+        Err(e) => Err(e.into()),
```
Comment (EvolveArt) [inline src/processing/spot.rs:298]: can't we just use question mark or `.map_err` in the assignment thus avoiding this `match` operator

File: src/processing/spot.rs [hunk 2]
```diff
@@ -217,53 +218,83 @@ pub async fn process_long_tail_asset(
...
+    for source in sources.iter() {
+        log::info!("Processing data for pair: {} and source: {}", pair, source);
+        let (timestamp, price_deviation) =
+            get_price_deviation_for_source_from_chain(pool.clone(), &pair, source, decimals)
+                .await?;
+        timestamps.push(timestamp);
+        deviations.push(price_deviation);
+    }
+
+    let threshold = get_long_tail_threshold(&pair, sources.len()).unwrap();
```
Comment (EvolveArt) [inline src/processing/spot.rs:237]: no unwrap

File: src/processing/spot.rs [hunk 3]
```diff
@@ -217,53 +218,83 @@ pub async fn process_long_tail_asset(
...
+
+    // TODO: Maybe we should only consider recent sources, i.e in the last hour?
```
Comment (EvolveArt) [inline src/processing/spot.rs:239]: that's something we could consider at the config level tbh

File: src/processing/spot.rs [hunk 4]
```diff
@@ -217,53 +218,83 @@ pub async fn process_long_tail_asset(
...
+
+    Ok(timestamps.last().copied().unwrap())
+}
+
+pub async fn get_price_deviation_for_source_from_chain(
+    pool: deadpool::managed::Pool<AsyncDieselConnectionManager<AsyncPgConnection>>,
+    pair: &str,
+    source: &str,
+    decimals: u32,
+) -> Result<(u64, f64), MonitoringError> {
```
Comment (EvolveArt) [inline src/processing/spot.rs:264]: let's return a struct here 
```suggestion
) -> Result<Deviation, MonitoringError> {
```

File: src/config.rs [hunk 5]
```diff
@@ -451,10 +451,28 @@ async fn init_future_config(
...
+
+#[allow(dead_code)]
+/// Retrieves the long tail asset threshold configuration depending on the number of sources.
+pub fn get_long_tail_threshold(pair: &str, number_of_sources: usize) -> Option<f64> {
+    if !is_long_tail_asset(pair) {
+        return None;
+    };
+    let (threshold_low, threshold_high) = LONG_TAIL_ASSETS.get(pair).unwrap();
+
+    if number_of_sources <= 6 {
```
Comment (EvolveArt) [inline src/config.rs:472]: no magic constant

File: src/processing/spot.rs (hunk 2 above)
Comment (EvolveArt) [reply src/processing/spot.rs:237]: in that case just add a comment on top explaining why it's fine

File: src/processing/spot.rs (hunk 1 above)
Comment (akhercha) [reply src/processing/spot.rs:298]: ah yeah just copied the already existing code - updated that everywhere

File: src/processing/spot.rs (hunk 2 above)
Comment (akhercha) [reply src/processing/spot.rs:218]: done

File: src/processing/spot.rs (hunk 4 above)
Comment (akhercha) [reply src/processing/spot.rs]: done

**Pull request merged**